packages/vc-verification/src/resolver/Ipfs-credential-resolver.ts

```diff
@@ -57,10 +59,20 @@ export class IpfsCredentialResolver implements CredentialResolver {
       | VerifiableCredential<RoleCredentialSubject>
       | RoleEIP191JWT
       | undefined;
+    const cachedRoleCredential = this._roleCredentialCache?.getRoleCredential(
+      did,
+      namespace
+    );
+    if (cachedRoleCredential) {
+      return cachedRoleCredential;
+    }
     credential = await this.getVerifiableCredential(did, namespace);
     if (!credential) {
       credential = await this.getEIP191JWT(did, namespace);
     }
+    if (credential) {
+      this._roleCredentialCache?.setRoleCredential(did, namespace, credential);
+    }
```

Comment:
Thinking about this some more, I think this cache might be ineffective. 
It seems to be the case for each `did`+`namespace` combination, `this.credentialsOf(did)` will be called.
Maybe it makes sense to cache all of the credentials resolved for a given `did`? Though this would increase the cache size, which we might not want.
What are your thoughts here @nichonien ?

Reply:
@jrhender If you look at this method signature it does have both `did` and `namespace`, so it can very well pull credentials from cache. While the `this.credentialsOf(did)` is only called to fetch all the VCs and then it is processed to return VC specific to the namespace. Also, the first call will always be for `getCredential` and it is much convenient to pull cache data here itself. Am I missing something or not understanding you concern?

Reply:
My concern, in summary, is that there would be multiple calls to `credentialsOf` for the same DID to retrieve the same information if there are multiple calls to `getCredentials` for different DID+namespace combinations. Do you see this as a problem?

Reply:
Nope, the credential resolution will never go to `credentialsOf` if we have already cached the role credential and thus `getCredential` will return without moving forward. The issuer-verification code relies only on `getCredential` method.

Reply:
> Nope, the credential resolution will never go to credentialsOf if we have already cached the role credential

But if you request a different credential for the same DID, then it will go to `credentialsOf` again, right? That is my concern.

Reply:
Hmm, then I think a better way to do it would be to cache the DID Document and process it for newer credentials. What do you think?

Reply:
I think it depends on how we anticipate the cache being used. The cache for `IpfsCredentialResolver` is currently optimized for resolving the same role credential for the same DID multiple times. Is this the use case that `IpfsCredentialResolver` is used for for issuer and revoker verification?